packages/site-parsers/src/parsers/wix/components/tpa-widget.js

```diff
@@ -0,0 +1,21 @@
+const { parseComponent: parseSpotify } = require( './spotify' );
+
+const APP_ID = {
+	SPOTIFY: '2575',
+	WIX_MUSIC: '1662',
+	SOUND_CLOUD: '3195',
+};
+
+module.exports = {
+	type: 'TPAWidget',
+	parseComponent: ( component ) => {
+		switch ( component.dataQuery.applicationId ) {
+			case APP_ID.SPOTIFY:
+				return parseSpotify( component );
+			case APP_ID.WIX_MUSIC:
+			case APP_ID.SOUND_CLOUD:
```

Comment:
While we're here, let's add in support for SoundCloud now, since Core supports it. 🙂

Reply:
You are right. :)

I've added the SoundCloud component mapper. Since the additional data is not a part of the `document_data` field, I've created a method `getThemeDataRef` which gives us an object from the `theme_data` field. In this case, fetching additional SoundCloud data by `styleId`.

Reply:
Noice, thanks for doing that!